packages/outbox-prisma-adapter/lib/outbox-prisma-adapter.ts

```diff
@@ -0,0 +1,167 @@
+import type {
+  OutboxAccumulator,
+  OutboxEntry,
+  OutboxStorage,
+} from '@message-queue-toolkit/outbox-core'
+import { type CommonEventDefinition, getMessageType } from '@message-queue-toolkit/schemas'
+import type { PrismaClient } from '@prisma/client'
+
+type ModelDelegate = {
+  // biome-ignore lint/suspicious/noExplicitAny: <explanation>
+  create: (args: any) => Promise<any>
+  // biome-ignore lint/suspicious/noExplicitAny: <explanation>
+  findMany: (args: any) => Promise<any>
+  // biome-ignore lint/suspicious/noExplicitAny: <explanation>
+  createMany: (args: any) => Promise<any>
+  // biome-ignore lint/suspicious/noExplicitAny: <explanation>
+  updateMany: (args: any) => Promise<any>
+}
+
+export class OutboxPrismaAdapter<
+  SupportedEvents extends CommonEventDefinition[],
+  ModelName extends keyof PrismaClient & string,
+> implements OutboxStorage<SupportedEvents>
+{
+  constructor(
+    private readonly prisma: PrismaClient,
+    private readonly modelName: ModelName,
+  ) {}
+
+  createEntry(
+    outboxEntry: OutboxEntry<SupportedEvents[number]>,
+  ): Promise<OutboxEntry<SupportedEvents[number]>> {
+    const prismaModel = this.prisma[this.modelName] as unknown as ModelDelegate
+
+    // @ts-ignore
```

Comment:
TODO: remove it later, added now to just move forward with the CI and setting up tests there.